x-pack/test/functional/apps/discover/async_scripted_fields.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// Tests for scripted field in default distribution where async search is used
+import expect from '@kbn/expect';
+
+export default function ({ getService, getPageObjects }) {
+  const kibanaServer = getService('kibanaServer');
+  // const log = getService('log');
+  const retry = getService('retry');
+  const esArchiver = getService('esArchiver');
+  const log = getService('log');
+  const testSubjects = getService('testSubjects');
+
+  const PageObjects = getPageObjects(['common', 'settings', 'discover', 'timePicker']);
+  const queryBar = getService('queryBar');
+
+  describe('async search with scripted fields', function () {
+    this.tags(['skipFirefox']);
+
+    before(async function () {
+      await esArchiver.load('kibana_scripted_fields_on_logstash');
+      await esArchiver.loadIfNeeded('logstash_functional');
+      // changing the timepicker default here saves us from having to set it in Discover (~8s)
+      await kibanaServer.uiSettings.update({
+        'timepicker:timeDefaults':
+          '{  "from": "Sep 18, 2015 @ 19:37:13.000",  "to": "Sep 23, 2015 @ 02:30:09.000"}',
+      });
+    });
+
+    after(async function afterAll() {
+      await kibanaServer.uiSettings.replace({});
+      await kibanaServer.uiSettings.update({});
```

Comment:
this is just a question, there is no way to replace a single uiSetting ? just asking because of https://github.com/elastic/kibana/blob/3ee0bf21324841a80e09cb3606b8f9f4887b550d/test/functional/page_objects/time_picker.ts#L64